### PR TITLE
Added getByID and searchByName

### LINKED ISSUE #103: Get user by mail
Get a user object using the user-service by its mail (which is currently his id).

### LINKED ISSUE #104: Add Get name autocomplete in users api

### PR DESCRIPTION
**Pull request by Shahar-Y**
Tested with Postman.
Closes #103 (Decided to not implement, GetByID implemented instead)
Closes #104

### CODE REVIEW

File: server/router.go [hunk 1]
```diff
@@ -120,7 +130,7 @@ func NewRouter(logger *logrus.Logger) (*gin.Engine, []*grpc.ClientConn) {
 	pr.Setup(authRequiredRoutesGroup)
 
 	// Create a slice to manage connections and return it.
-	return r, []*grpc.ClientConn{fileConn, uploadConn, downloadConn, permissionConn}
+	return r, []*grpc.ClientConn{fileConn, uploadConn, downloadConn, userConn, permissionConn}
```
Comment (omrishtam) [inline server/router.go]: Put `userConn` at the end.

File: user/user.go [hunk 2]
```diff
@@ -19,6 +34,75 @@ type User struct {
...
+
+// Setup sets up r and intializes its routes under rg.
+func (r *Router) Setup(rg *gin.RouterGroup) {
+	rg.GET("/users/:id", r.GetUserByID)
+	rg.GET("/users", r.SearchByName)
+}
+
+// GetUserByID is the request handler for GET /users/:id
+func (r *Router) GetUserByID(c *gin.Context) {
+	userID := c.Param("id")
```
Comment (omrishtam) [inline user/user.go]: Check that there's an authenticated user in the context of the request like in `file.Router`

File: user/user.go [hunk 3]
```diff
@@ -19,6 +34,75 @@ type User struct {
...
+
+// SearchByName is the request handler for GET /users
+func (r *Router) SearchByName(c *gin.Context) {
+	partialName := c.Query("partial")
+	if partialName == "" {
+		c.String(http.StatusBadRequest, "partial name required")
+		return
+	}
+
+	FindUserByNameRequest := &uspb.FindUserByNameRequest{
```
Comment (omrishtam) [inline user/user.go]: Should be cammel case `findUserByNameRequest`

File: user/user.go [hunk 4]
```diff
@@ -19,6 +34,75 @@ type User struct {
...
+		Name: partialName,
+	}
+
+	user, err := r.userClient.FindUserByName(c.Request.Context(), FindUserByNameRequest)
+
+	if err != nil {
+		httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
+		loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
```
Comment (omrishtam) [inline user/user.go]: Missing `return` after error handling.

File: user/user.go [hunk 5]
```diff
@@ -19,6 +34,75 @@ type User struct {
...
+
+	getUserByIDRequest := &uspb.GetByIDRequest{
+		Id: userID,
+	}
+
+	user, err := r.userClient.GetUserByID(c.Request.Context(), getUserByIDRequest)
+
+	if err != nil {
+		httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
+		loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
```
Comment (omrishtam) [inline user/user.go]: Missing `return` after error handling.

File: user/user.go (hunk 2 above)
Comment (omrishtam) [inline user/user.go]: Use constant for param `id` like done in `permission` package, and its routes with `fmt.Sscanf`.

**Comment by Shahar-Y** [comment]
@omrishtam fixed and tested

**Pull request merged**